libraries/royalties-interface/src/lib.rs

```diff
@@ -0,0 +1,103 @@
+use solana_sdk::{
+    account_info::AccountInfo, instruction::Instruction, program_error::ProgramError,
+    pubkey::Pubkey, system_instruction::transfer,
+};
+use spl_associated_token_account::{
+    get_associated_token_address_with_program_id,
+    instruction::create_associated_token_account_idempotent,
+};
+use spl_token_2022::{
+    extension::{BaseStateWithExtensions, StateWithExtensions},
+    instruction::transfer_checked,
+    state::Mint,
+};
+use spl_token_metadata_interface::state::TokenMetadata;
+use std::str::FromStr;
+
+pub const ROYALTY_BASIS_POINTS_FIELD: &str = "royalty_basis_points";
+
+pub fn calculate_royalties(
+    mint: &AccountInfo,
+    amount: u64,
+) -> Result<(u64, TokenMetadata, Mint), ProgramError> {
+    let mint_account_data = mint.try_borrow_data()?;
+    let mint_data = StateWithExtensions::<Mint>::unpack(&mint_account_data)?;
+    let metadata = mint_data.get_variable_len_extension::<TokenMetadata>()?;
+
+    // get royalty basis points from metadata Vec(String, String)
+    let royalty_basis_points = metadata
+        .additional_metadata
+        .iter()
+        .find(|(key, _)| key == ROYALTY_BASIS_POINTS_FIELD)
+        .map(|(_, value)| u64::from_str(value).unwrap())
```

Comment:
```suggestion
        .map(|(_, value)| u64::from_str(value))?
```

maybe not quite this but should be able to handle the result without `unwrap`